File: Sakila_Net/Sakila_Net/VB_SakilaLib/Model/Address.vb
Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace Model
public class ClAddress

   private addressId as Integer
   private address as String
   private address2 as String
   private district as String
   private  cityId  as ClCity
   private postalCode as String
   private phone as String
   private lastUpdate as Date


   public Property DataAddressId() as Integer
        Get
            Return addressId
        End Get
        Set(ByVal Value as Integer)
            addressId = Value
        End Set        
   End Property   
   public Property DataAddress() as String
        Get
            Return address
        End Get
        Set(ByVal Value as String)
            address = Value
        End Set        
   End Property   
   public Property DataAddress2() as String
        Get
            Return address2
        End Get
        Set(ByVal Value as String)
            address2 = Value
        End Set        
   End Property   
   public Property DataDistrict() as String
        Get
            Return district
        End Get
        Set(ByVal Value as String)
            district = Value
        End Set        
   End Property   
   public Property DataCityId()  as ClCity
        Get
            Return cityId
        End Get
        Set(ByVal Value  as ClCity)
            cityId = Value
        End Set        
   End Property   
   public Property DataPostalCode() as String
        Get
            Return postalCode
        End Get
        Set(ByVal Value as String)
            postalCode = Value
        End Set        
   End Property   
   public Property DataPhone() as String
        Get
            Return phone
        End Get
        Set(ByVal Value as String)
            phone = Value
        End Set        
   End Property   
   public Property DataLastUpdate() as Date
        Get
            Return lastUpdate
        End Get
        Set(ByVal Value as Date)
            lastUpdate = Value
        End Set        
   End Property   

   Public Sub New (addressId    as Integer,address    as String,address2    as String,district    as String,cityId  as ClCity,postalCode    as String,phone    as String,lastUpdate    as Date)
		   dataAddressId = addressId
		   dataAddress = address
		   dataAddress2 = address2
		   dataDistrict = district
		   dataCityId = cityId
		   dataPostalCode = postalCode
		   dataPhone = phone
		   dataLastUpdate = lastUpdate
   End Sub
   Public Sub New (addressId    as Integer)
    dataAddressId = addressId
   End Sub


   public Overrides Function  ToString() As String
        '// Registro JSon
		return "{" _ 
		    & "," & chr(34) & "addressId" & chr(34) & ":" & chr(34) & dataAddressId & chr(34) _
		    & "," & chr(34) & "address" & chr(34) & ":" & chr(34) & dataAddress & chr(34) _
		    & "," & chr(34) & "address2" & chr(34) & ":" & chr(34) & dataAddress2 & chr(34) _
		    & "," & chr(34) & "district" & chr(34) & ":" & chr(34) & dataDistrict & chr(34) _
		    & "," & chr(34) & "cityId" & chr(34) & ":" & chr(34) & dataCityId.ToString() & chr(34) _
		    & "," & chr(34) & "postalCode" & chr(34) & ":" & chr(34) & dataPostalCode & chr(34) _
		    & "," & chr(34) & "phone" & chr(34) & ":" & chr(34) & dataPhone & chr(34) _
		    & "," & chr(34) & "lastUpdate" & chr(34) & ":" & chr(34) & dataLastUpdate & chr(34) _
		 & "}"
   End Function   
   public Sub imprimir()
        Console.WriteLine(ToString())
   End Sub
   public Function jsonSerialize()
           return "get_object_vars($this);"
   End Function   
End Class
         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Php/Entities/Address.php
<?PHP
class Address implements JsonSerializable{

   private $addressId;
   private $address;
   private $address2;
   private $district;
   private $postalCode;
   private $phone;
   private $lastUpdate;


   public function getAddressId(){
        return $this->addressId;
   }
   public function setAddressId($addressId){
        $this->addressId = $addressId;
   }
   
   public function getAddress(){
        return $this->address;
   }
   public function setAddress($address){
        $this->address = $address;
   }
   
   public function getAddress2(){
        return $this->address2;
   }
   public function setAddress2($address2){
        $this->address2 = $address2;
   }
   
   public function getDistrict(){
        return $this->district;
   }
   public function setDistrict($district){
        $this->district = $district;
   }
   
   public function getCityId(){
        return $this->cityId;
   }
   public function setCityId($cityId){
        $this->cityId = $cityId;
   }
   
   public function getPostalCode(){
        return $this->postalCode;
   }
   public function setPostalCode($postalCode){
        $this->postalCode = $postalCode;
   }
   
   public function getPhone(){
        return $this->phone;
   }
   public function setPhone($phone){
        $this->phone = $phone;
   }
   
   public function getLastUpdate(){
        return $this->lastUpdate;
   }
   public function setLastUpdate($lastUpdate){
        $this->lastUpdate = $lastUpdate;
   }
   

   public function Address($addressId,$address='',$address2='',$district='',$cityId='',$postalCode='',$phone='',$lastUpdate=''){
		   $this->setAddressId($addressId);
		   $this->setAddress($address);
		   $this->setAddress2($address2);
		   $this->setDistrict($district);
		   $this->setCityId($cityId);
		   $this->setPostalCode($postalCode);
		   $this->setPhone($phone);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __constructXX($addressId,$address='',$address2='',$district='',$cityId='',$postalCode='',$phone='',$lastUpdate='') {
		   $this->setAddressId($addressId);
		   $this->setAddress($address);
		   $this->setAddress2($address2);
		   $this->setDistrict($district);
		   $this->setCityId($cityId);
		   $this->setPostalCode($postalCode);
		   $this->setPhone($phone);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __toString(){
        // Registro JSon
		return "{" 
		    . "," . chr(34) . "addressId" . chr(34) . ":" . chr(34) . $this->getAddressId() . chr(34) 
		    . "," . chr(34) . "address" . chr(34) . ":" . chr(34) . $this->getAddress() . chr(34) 
		    . "," . chr(34) . "address2" . chr(34) . ":" . chr(34) . $this->getAddress2() . chr(34) 
		    . "," . chr(34) . "district" . chr(34) . ":" . chr(34) . $this->getDistrict() . chr(34) 
		    . "," . chr(34) . "cityId" . chr(34) . ":" . chr(34) . $this->getCityId() . chr(34) 
		    . "," . chr(34) . "postalCode" . chr(34) . ":" . chr(34) . $this->getPostalCode() . chr(34) 
		    . "," . chr(34) . "phone" . chr(34) . ":" . chr(34) . $this->getPhone() . chr(34) 
		    . "," . chr(34) . "lastUpdate" . chr(34) . ":" . chr(34) . $this->getLastUpdate() . chr(34) 
		 . "}";
   }   
   public function imprimir()
   {
        echo $this->__toString();
   }
   public function jsonSerialize()
   {
        return get_object_vars($this);
   }   
}
// Para realizar pruebas
//$actor = new Address();
//echo "Imprimiendo ";$actor->imprimir();
//echo "ToString : " . $actor;
?>
